tests/lib/prepare.sh

```diff
@@ -154,6 +152,25 @@ setup_experimental_features() {
     fi
 }
 
+save_installed_core_snap() {
+    local target_dir="${1-}"
+
+    SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    LIBEXEC_DIR="$(os.paths libexec-dir)"
+    core="$(readlink -f "$SNAP_MOUNT_DIR/core/current")"
+    snap="$(mount | grep " $core" | head -n 1 | awk '{print $1}')"
```

Comment:
I think the logic here is pretty weak. `$core` is just the revision number and maybe there are other snaps that randomly have the same revision.

Perhaps something like this:
```sh
mount | awk '{ if ($3 ~ /\/snap\/core\/17201$/) print $1 }'
```

Note that you have to adjust this so that core revision and mount path is coming from the earlier line.

Reply:
this is the same logic we are using in update_core_snap_for_classic_reexec 
Updated both

Reply:
Thank you!